memstore/archiving.go

```diff
@@ -138,12 +141,24 @@ func (ss liveStoreSnapshot) createArchivingPatches(
 func (s *LiveStore) getBatchIDsToPurge(cutoff uint32) []int32 {
 	var batchIDs []int32
 	s.RLock()
+
+	s.tableSchema.RLock()
+	allowMissingEventTime := s.tableSchema.Schema.Config.AllowMissingEventTime
+	s.tableSchema.RUnlock()
+
 	for batchID, batch := range s.Batches {
 		if batchID >= s.LastReadRecord.BatchID {
 			continue
 		}
 
-		if _, maxValue := batch.Columns[0].(common.LiveVectorParty).GetMinMaxValue(); maxValue < cutoff {
+		timeColumn := batch.Columns[0]
+		if timeColumn != nil {
+			if _, maxValue := timeColumn.(common.LiveVectorParty).GetMinMaxValue(); maxValue < cutoff {
+				if !allowMissingEventTime || timeColumn.GetNonDefaultValueCount() == batch.Capacity || batch.MaxArrivalTime < cutoff {
```

Comment:
how the maxArrivalTime is chosen and updated? the no event timestamp event maybe updated with real timestamp, while some older event still has no timestamp. maybe no big issue